fontforge/splinestroke.c

```diff
@@ -265,6 +266,11 @@ static int LineSameSide(BasePoint l1, BasePoint l2, BasePoint p, BasePoint r,
 
     if ( RealWithin(tp, 0, 1e-5) )
 	return on_line_ok;
+    // Rounding means we should be tolerant when on_line_ok is true.
+    // Being that tolerant when on_line_ok is false would 
+    // yield too many false negatives. 
+    if ( on_line_ok && RealWithin(tp, 0, 1) )
```

Comment:
Seems like a pretty big jump in tolerance?

Reply:
right, so it's effectively

```c
if ((on_line_ok && RealWithin(tp, 0, 1)) || !RealWithin(tp, 0, 1e-5))
    return true;
```

Reply:
ok not quite.

Reply:
The intention is a looser margin for the `on_line_ok` is true case and a tighter one for `on_line_ok` is false. That way if being on the line is "not allowed" you can get within the tighter margin of it and still be "ok".